src/__tests__/dragon.test.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import configureMockStore from 'redux-mock-store';
+import Dragon from '../components/dragons/Dragon';
+import '@testing-library/jest-dom';
+
+const mockStore = configureMockStore([]);
+
+describe('Dragon component', () => {
+  let store;
+
+  beforeEach(() => {
+    store = mockStore({
+      dragons: {
+        dragons: [
+          {
+            id: '1',
+            name: 'Dragon 1',
+            type: 'Type 1',
+            description: 'A cool dragon',
+            image: 'http://example.com/dragon1.jpg',
+            isReserved: false,
+          },
+          {
+            id: '2',
+            name: 'Dragon 2',
+            type: 'Type 2',
+            description: 'An even cooler dragon',
+            image: 'http://example.com/dragon2.jpg',
+            isReserved: false,
+          },
+        ],
+      },
+    });
+
+    store.dispatch = jest.fn();
+  });
+
+  it('should dispatch fetchDragons action on mount', () => {
+    render(
+      <Provider store={store}>
+        <Dragon />
+      </Provider>,
+    );
+
+    expect(store.dispatch).toEqual(expect.any(Function));
+  });
+  it('should dispatch bookDragons action on reserve button click', () => {
+    // Test case for bookDragons action creator
```

Comment:
- [ ] [OPTIONAL] I think you are missing some test case here, and also try to test your reducers as they are pure functions, maybe you can test them for each of the components, as currently you are only testing render part.